app/src/main/java/dracula/vlad/androcalc/MainActivity.java

```diff
@@ -27,81 +27,68 @@
 
 public class MainActivity extends AppCompatActivity {
 
-    private String TAG = "LOG_Main";
-    private String input = "";
-    private ArrayList<Button> buttonArrayList= new ArrayList<>();
-    private double result;
-    TextView tv_Res=null;
+    private static final String TAG = MainActivity.class.getName();
+    private static final String DIVISION = "/";
+    private static final String MULTIPLICATION = "*";
+    private static final String ADDITION = "+";
+    private static final String SUBTRACTION = "-";
+    private static final String EQUATION = "=";
+    public static final String DOT = ".";
+
+    private String userInput;
+    private ArrayList<Button> buttons = new ArrayList<>();
+    private double userResult;
+
+    private TextView resultView;
+    private EditText inputView;
+    private Button clearAllButton;
+    private Button deleteSingleCharButton;
+    private Button dotInputButton;
 
-    @SuppressLint("ClickableViewAccessibility")
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        getChildViews(findViewById(R.id.ll_Buttons));
-
-        final EditText editText = findViewById(R.id.et_Input);
-        editText.requestFocus();
+        getChildViews(findViewById(R.id.buttons_linear_layout));
 
-        tv_Res = findViewById(R.id.tv_Result);
+        resultView = findViewById(R.id.result_text_view);
+        inputView = findViewById(R.id.user_input_edit_text);
+        inputView.requestFocus();
 
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
-            editText.setShowSoftInputOnFocus(false);
-        }
-        else{
-            editText.setOnTouchListener((v, event) -> {
-                v.onTouchEvent(event);
-                InputMethodManager inputMethod = (InputMethodManager) v.getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
-                if (inputMethod != null) {
-                    inputMethod.hideSoftInputFromWindow(v.getWindowToken(), 0);
-                }
-                return true;
-            });
-        }
+        hideSoftwareKeyboard();
```

Comment:
in current project it is ok but for the future purpose think about extracting it to utils class or Kotlin extension etc